homeassistant/components/zone/__init__.py

```diff
@@ -333,6 +345,40 @@ async def async_update_config(self, config: dict) -> None:
         self._generate_attrs()
         self.async_write_ha_state()
 
+    @callback
+    def _person_state_change_listener(self, evt: Event) -> None:
+        object_id = split_entity_id(self.entity_id)[1]
+        person_entity_id = evt.data["entity_id"]
+        if evt.data["new_state"] and evt.data["new_state"].state == object_id:
+            self._persons_in_zone.add(person_entity_id)
+            self.async_write_ha_state()
+        elif person_entity_id in self._persons_in_zone:
+            self._persons_in_zone.remove(person_entity_id)
+            self.async_write_ha_state()
```

Comment:
Now every zone will do a write state if a person changs it's state regardless if the zone actually changed it's state.

```suggestion
        cur_count = len(self._persons_in_zone)
        if evt.data["new_state"] and evt.data["new_state"].state == object_id:
            self._persons_in_zone.add(person_entity_id)
        elif person_entity_id in self._persons_in_zone:
            self._persons_in_zone.remove(person_entity_id)

        if len(self._persons_in_zone) != cur_count:
            self.async_write_ha_state()
```